stumpy/fastmath.py

```diff
@@ -0,0 +1,89 @@
+import ast
+import importlib
+import pathlib
+
+from stumpy import config
+
+
+def get_default_fastmath(module_name):
+    """
+    Retrieve a dictionary where key is njit function name
+    and value is the default fastmath flag.
+
+    Parameters
+    ----------
+    module_name : str
+        The module name
+
+    Returns
+    -------
+    out : dict
+        A dictionary of njit functions, where key is the function name
+        and value is the fastmath flag.
+    """
+    filepath = pathlib.Path(__file__).parent / f"{module_name}.py"
+    file_contents = ""
+    with open(filepath, encoding="utf8") as f:
+        file_contents = f.read()
+
+    out = {}
+    module = ast.parse(file_contents)
+    for node in module.body:
+        if not isinstance(node, ast.FunctionDef):
+            continue
+
+        func_name = node.name
+        for decorator in node.decorator_list:
+            if not isinstance(decorator, ast.Call) or not isinstance(
+                decorator.func, ast.Name
+            ):
+                continue
+
+            if decorator.func.id != "njit":
+                continue
+
+            fastmath_default = None
+            for item in decorator.keywords:
+                if item.arg == "fastmath":
+                    config_var = item.value.attr
+                    fastmath_default = config._STUMPY_DEFAULTS[config_var]
+                    break
+
+            out[func_name] = fastmath_default
+
+    return out
+
+
+def set_flag(module_name, func_name, flag=None):
```

Comment:
When `flag=None`, this function resets a fastmath to its default value. Initially, I thought it would be cool to support both set & reset in a single function. Now I think that it will be cleaner if we separate them.

```
def set_flag(module_name, func_name, flag):
       # and raise error if flag is None
       # set flag & recompile


def reset_flag(module_name, func_name):
     # get default_flag
     # call set_flag(module_name, func_name, flag=default_flag):
```

Reply:
I think the functions should be `fastmath._set` and `fastmath._reset` (i.e., `flag` is not needed. It is good to split them. Based on my comment above, we can then do:

```
def _reset(module_name, func_name):
    key = module_name + '.' + func_name
    key = 'STUMPY_' + key.upper()
    flags = config._STUMPY_DEFAULTS[key]
    _set(module_name, func_name, flags)
```

This is really clean!

Reply:
What's also nice is that `flags = config._STUMPY_DEFAULTS[key]` immediately indicates WHERE the default value is coming from and likely where it is being set initially!